### ISSUE TITLE
Label 11 too scarce in `Books/history` dataset causing split inconsistency

### ISSUE THREAD

**Issue by Cloudy1225** [WxxShirley/LLMNodeBed#28]
Hi, I encountered an issue when working with the `Books/history` dataset.

It seems that label 11 has only 4 samples, which causes an imbalance during dataset splitting. Specifically, label 11 appears *only* in the test set and not in the train/val sets, leading to potential issues in model evaluation and generalization.

Here’s the code I used to inspect the split:

```python
dataset = 'history'
encoder_name = 'roberta'
re_split = 0
data = load_graph_dataset_for_gnn(dataset_name=dataset,
                                  device='cpu',
                                  path_prefix='.',
                                  emb_model=encoder_name if len(encoder_name) else "shallow",
                                  re_split=re_split)
y, train_mask, val_mask, test_mask = data.y, data.train_mask, data.val_mask, data.test_mask
print(y[train_mask].unique())
print(y[val_mask].unique())
print(y[test_mask].unique())
print(sorted([(y==l).sum().item() for l in y.unique()]))
```

Output:

```
tensor([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10])
tensor([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10])
tensor([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11])
[4, 64, 96, 250, 280, 307, 747, 876, 3360, 5304, 6895, 23368]
```

As shown above, label 11 is entirely missing from the training and validation sets, which likely affects performance.

**Comment by WxxShirley** [comment]
Thank you for highlighting the data splits in the Books-History dataset.

It was unexpected for one category to appear only in the test set. We randomly divided the data into training, validation, and test sets in a 1:1:6 ratio for the semi-supervised setting and a 6:2:2 ratio for the supervised setting.

Since **all methods are evaluated under the same conditions**, the evaluation remains fair.
